pkg/cmd/roachtest/testselector/selector_test.go

```diff
@@ -0,0 +1,159 @@
+	})
+	var mock sqlmock.Sqlmock
+	var db *gosql.DB
+	SqlConnectorFunc = func(driverName, dataSourceName string) (*gosql.DB, error) {
+		dsn, err := getDSN()
+		require.Equal(t, "snowflake", driverName)
+		require.Equal(t, dsn, dataSourceName)
+		return db, err
+	}
+	var err error
+	t.Run("expect Prepare to fail", func(t *testing.T) {
+		db, mock, err = sqlmock.New()
+		require.Nil(t, err)
+		mock.ExpectPrepare(regexp.QuoteMeta(PreparedQuery)).WillReturnError(fmt.Errorf("failed to prepare"))
+		tds, err := CategoriseTests(context.Background(), nil)
+		require.Nil(t, tds)
+		require.NotNil(t, err)
+		require.Equal(t, "failed to prepare", err.Error())
+	})
+	t.Run("expect query to fail", func(t *testing.T) {
+		db, mock, err = sqlmock.New()
+		require.Nil(t, err)
+		mock.ExpectPrepare(regexp.QuoteMeta(PreparedQuery))
+		mock.ExpectQuery(regexp.QuoteMeta(PreparedQuery)).WillReturnError(fmt.Errorf("failed to execute query"))
+		tds, err := CategoriseTests(context.Background(), &SelectTestsReq{
+			ForPastDays: 1,
+			FirstRunOn:  2,
+			LastRunOn:   3,
+			Cloud:       spec.AWS,
+			Suite:       "unit test",
+		})
+		require.Nil(t, tds)
+		require.NotNil(t, err)
+		require.Equal(t, "failed to execute query", err.Error())
+	})
+	t.Run("expect no failure", func(t *testing.T) {
+		db, mock, err = sqlmock.New()
+		mock.ExpectPrepare(regexp.QuoteMeta(PreparedQuery))
+		rows := sqlmock.NewRows([]string{
+			"name", "selected", "avg_duration", "last_failure_is_preempt",
+		})
+		data := [][]string{
+			{"t1", "no", "12345", "no"},
+			{"t2", "no", "12345", "no"},
+			{"t3", "no", "12345", "yes"},
+			{"t4", "no", "12345", "no"},
+			{"t5", "yes", "12345", "no"},
+			{"t6", "yes", "12345", "no"},
+			{"t7", "yes", "12345", "no"},
+			{"t8", "yes", "12345", "no"},
+			{"t9", "yes", "12345", "no"},
+		}
+		for _, ds := range data {
+			rows.FromCSVString(strings.Join(ds, ","))
+		}
+		mock.ExpectQuery(regexp.QuoteMeta(PreparedQuery)).WillReturnRows(rows)
+		tds, err := CategoriseTests(context.Background(), &SelectTestsReq{
+			ForPastDays: 1,
+			FirstRunOn:  2,
+			LastRunOn:   3,
+			Cloud:       spec.AWS,
+			Suite:       "unit test",
+		})
+		require.NotNil(t, tds)
+		require.Nil(t, err)
+		require.Equal(t, len(data), len(tds))
+		// the sequence of response list must be maintained.
+		for i, d := range data {
+			td := tds[i]
+			require.Equal(t, d[0], td.Name)
+			require.Equal(t, d[1] != "no", td.Selected)
+			require.Equal(t, getDuration(d[2]), td.AvgDurationInMillis)
+			require.Equal(t, d[3] == "yes", td.LastFailureIsPreempt)
+		}
+	})
+}
+
+func TestNewDefaultSelectTestsReq(t *testing.T) {
+	req := NewDefaultSelectTestsReq(spec.Azure, "ut suite")
+	require.Equal(t, spec.Azure, req.Cloud)
+	require.Equal(t, "ut suite", req.Suite)
+	require.Equal(t, defaultForPastDays, req.ForPastDays)
+	require.Equal(t, defaultFirstRunOn, req.FirstRunOn)
+	require.Equal(t, defaultLastRunOn, req.LastRunOn)
+}
+
+func Test_getSFCreds(t *testing.T) {
```

Comment:
Similar comment to above, these tests seem to be already covered by `TestCategoriseTests`? Not sure how much extra coverage we get.

Reply:
This is for the coverage of all error scenarios and ensure that we are returning the errors.